File: Aula4-parte2/Em_java/Item.java
public class Item{
    private String nome;
    private int bonus;
    public Item(String nome, int bonus){
        this.nome = nome;
        this.bonus = bonus;
    }
    public String getNome(){
        return this.nome;
    }
    public void setNome(String nome){
        if(nome == null || nome.isEmpty()){
            System.out.println("ERRO, nome vazio\n");
        }else{
            this.nome = nome;
        }
    }
    public int getBonus(){
            return this.bonus;
    }
    public void setBonus(int bonus){
        if(bonus >= 0){
            this.bonus = bonus;
        }else{
            System.out.println("ERRO, bonus negativo\n");
        }
    }
    public String Descricao(){
        return "Item: " + this.nome + ", bonus: " + this.bonus;
    }
}

File: IF Quest/Em_python/Item.py
class Item:
    def __init__(self, nome: str, bonus: int):
        self._nome = nome
        self._bonus = bonus

    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, nome: str):
        if not nome:
            print("O nome vazio.")
        else:
            self._nome = nome

    @property
    def bonus(self):
        return self._bonus

    @bonus.setter
    def bonus(self, bonus: int):
        if bonus < 0:
            print("O bônus não pode ser negativo.")
        else:
            self._bonus = bonus

    def descricao(self):
        return self.__str__()

    def __str__(self):
        return f"Item: {self._nome}, Bônus: {self._bonus}"
